src/toolkit/bamtools_convert.cpp

```diff
@@ -381,6 +407,112 @@ void ConvertTool::ConvertToolPrivate::PrintFastq(const BamAlignment& a)
           << qualities << std::endl;
 }
 
+// Print out the list of values in a B-type tag.
+// tagData should point to the sub-type character (after the B).
+// tagDataLength is the number of bytes remaining in the buffer.
+// Returns the number of bytes consumed.
+std::size_t ConvertTool::ConvertToolPrivate::PrintBArrayValues(const char* tagData,
+                                                               std::size_t tagDataLength)
+{
+    // Need at least 5 bytes for type and array length
+    if (tagDataLength < 5) throw BadDataException("Incomplete array tag data");
+
+    if (m_settings->Format == FORMAT_SAM) {
+        // Print array type letter
+        switch (tagData[0]) {
+            case (Constants::BAM_TAG_TYPE_INT8):
+                m_out << "c,";
+                break;
+            case (Constants::BAM_TAG_TYPE_UINT8):
+                m_out << "C,";
+                break;
+            case (Constants::BAM_TAG_TYPE_INT16):
+                m_out << "s,";
+                break;
+            case (Constants::BAM_TAG_TYPE_UINT16):
+                m_out << "S,";
+                break;
+            case (Constants::BAM_TAG_TYPE_INT32):
+                m_out << "i,";
+                break;
+            case (Constants::BAM_TAG_TYPE_UINT32):
+                m_out << "I,";
+                break;
+            case (Constants::BAM_TAG_TYPE_FLOAT):
+                m_out << "f,";
+                break;
+            default:
+                throw BadDataException("Unknown B array type");
+        }
+    }
+
+    uint32_t arrayLength = BamTools::UnpackUnsignedInt(&tagData[1]);
+    std::size_t index = 1 + sizeof(uint32_t);
+    uint32_t i = 0;
+
+    // Print array values.
+    switch (tagData[0]) {
+        case (Constants::BAM_TAG_TYPE_INT8):
+            for (i = 0; i < arrayLength && index + sizeof(int8_t) <= tagDataLength;
+                 i++, index += sizeof(int8_t)) {
+                if (i > 0) m_out << ",";
+                m_out << static_cast<int16_t>(tagData[index]);
```

Comment:
```suggestion
                m_out << int(static_cast<int8_t>(tagData[index]));;
```
consistency with line 601